columnar/src/backend/columnar/columnar_tableam.c

```diff
@@ -1200,6 +1201,22 @@ TruncateAndCombineColumnarStripes(Relation rel, int elevel)
 		}
 	}
 
+	/*
+	 * We need to clear current proces (VACUUUM) status flag here. Why?
+	 * Current process has flag `PROC_IN_VACUUM` which is problematic because
+	 * we write here into metadata heap tables. If concurrent process
+	 * read page in which we inserted metadata tuples these tuples will be considered
+	 * DEAD and will be removed (in problematic scenarion). 
+	 * Concurrent process, when assigning RecentXmin will scan all active processes in
+	 * system but will NOT consider this process because of 
+	 * `PROC_IN_VACUUM` flag set. It looks that invalidating status flag here doesn't affect 
+	 * further execution.
+	 */
+	LWLockAcquire(ProcArrayLock, LW_EXCLUSIVE);
+	MyProc->statusFlags = 0;
```

Comment:
`statusFlags` is pg14+ and not compatible with pg13.

Reply:
Added support for pg13